### PR TITLE
feat: Configure mandatory optional tags

### PR DESCRIPTION
**Pull request by vivek-devtron**
# Description

This PR consist the changes of the new libraries

Fixes # [AB#1361](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/1361)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?
Tested by creating multiple tags

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by shivani170 — APPROVED** [review]

File: src/components/apiTokens/APITokenList.tsx [hunk 1]
```diff
@@ -59,9 +58,9 @@ function APITokenList({ tokenList, renderSearchToken, reload }: APITokenListType
                 documentationLinkText="View Documentation"
             >
                 <div className="icon-dim-16 fcn-9 ml-8 cursor">
-                    <Question />    
+                    <Question />
                 </div>
-                
+
```
Comment (shivani170) [inline src/components/apiTokens/APITokenList.tsx:63]: Please remove this extra line

File: src/components/app/config.tsx [hunk 2]
```diff
@@ -74,5 +74,3 @@ export const APP_STATUS = {
     Missing: 'Missing',
     Progressing : 'Progressing',
 }
-
-export const DEFAULT_TAG_DATA = { key: '', value: '', propagate: false, isInvalidKey: false, isInvalidValue: false }
```
Comment (shivani170) [inline src/components/app/config.tsx:78]: Is this removal has no impact area

File: src/components/app/create/validationRules.ts [hunk 3]
```diff
@@ -79,4 +79,4 @@ export class ValidationRules {
         }
         return { isValid: errorList.length === 0, messages: errorList }
     }
-}
```
Comment (shivani170) [inline src/components/app/create/validationRules.ts:82]: Please revert this change

File: src/components/app/details/testViewer/TestRunList.tsx [hunk 4]
```diff
@@ -1,10 +1,11 @@
 import React, {useState, useMemo} from 'react'
-import {useAsync, Progressing, multiSelectStyles, mapByKey, Td, DropdownIcon, Option, noop, DatePicker} from '../../../common'
+import {useAsync, mapByKey, Td, DropdownIcon, DatePicker} from '../../../common'
```
Comment (shivani170) [inline src/components/app/details/testViewer/TestRunList.tsx:2]: DatePicker is in common or library ?

File: src/components/app/details/triggerView/ciMaterial.tsx [hunk 5]
```diff
@@ -1,15 +1,15 @@
 import React, { Component } from 'react'
+import { showError, Progressing, VisibleModal, ServerErrors, Checkbox } from '@devtron-labs/devtron-fe-common-lib'
```
Comment (shivani170) [inline src/components/app/details/triggerView/ciMaterial.tsx:2]: Are we using ServerErrors in this component?

Comment (vivek-devtron) [reply src/components/app/details/triggerView/ciMaterial.tsx:2]: yes

File: src/components/app/details/testViewer/TestRunList.tsx (hunk 4 above)
Comment (vivek-devtron) [reply src/components/app/details/testViewer/TestRunList.tsx:2]: it is in common

File: src/components/app/create/validationRules.ts (hunk 3 above)
Comment (vivek-devtron) [reply src/components/app/create/validationRules.ts:82]: it is due to the formatting

File: src/components/app/config.tsx (hunk 2 above)
Comment (vivek-devtron) [reply src/components/app/config.tsx:78]: it has been moved to the common lib

**Review by shivani170 — APPROVED** [review]

**Review by sohel-rp — APPROVED** [review]
